packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -319,42 +447,115 @@ const Combobox = forwardRef<HTMLDivElement, ComboboxProps>(
         onChange?.(event);
         // istanbul ignore else
         if (!isControlled) {
-          setValue(event.target.value);
+          setInputValue(event.target.value);
         }
       },
       [isControlled, onChange]
     );
 
-    const handleSelectionChange = useCallback(
-      ({
-        target,
-        value: listboxValue,
-        previousValue
-      }: Parameters<
-        Exclude<
-          React.ComponentProps<typeof Listbox>['onSelectionChange'],
-          undefined
-        >
-      >[0]) => {
-        const stringValue =
-          listboxValue?.toString() || /* istanbul ignore next */ '';
-
-        // istanbul ignore else
-        if (!isControlled) {
-          setValue(stringValue);
-        }
+    const handleRemove = useCallback(
+      (target: HTMLElement, value: ComboboxValue) => {
+        if (disabled) return;
 
-        setSelectedValue(stringValue);
+        const newSelectedValues = selectedValues.filter((v) => v !== value);
 
-        onSelectionChange?.({
+        setSelectedValues(newSelectedValues);
+        (onSelectionChange as OnMultiSelectionChange)?.({
           target,
-          value: stringValue,
-          previousValue: previousValue?.toString()
+          value: newSelectedValues,
+          previousValue: selectedValues
         });
+      },
+      [disabled, selectedValues, onSelectionChange]
+    );
+
+    const handlePillKeyDown = useCallback(
+      (event: React.KeyboardEvent<HTMLButtonElement>) => {
+        if (!PillKeys.includes(event.key)) {
+          return;
+        }
+
+        event.preventDefault();
+
+        const deleteKeypress = event.key === Delete;
+        const backspaceKeypress = event.key === Backspace;
+
+        const focusedIndex = pillsRef.current.findIndex(
+          (p) => document.activeElement === p
+        );
+
+        const focusedPill = pillsRef.current[focusedIndex];
+
+        if (deleteKeypress || backspaceKeypress) {
+          if (disabled) return;
```

Comment:
Nit: we generally don't use one line if statements.

```suggestion
          if (disabled) {
            return;
          }
```